pipeline/splicing/calcVarPriors.py

```diff
@@ -667,15 +686,305 @@ def getRefAltScores(refSeq, altSeq, donor=False):
+        slidingWindowInfo = getMaxMaxEntScanScoreSlidingWindowSNS(variant)
+        newSplicePos = getNewSplicePosition(variant["Pos"], getVarStrand(variant),
+                                            slidingWindowInfo["varWindowPosition"], slidingWindowInfo["inFirstThree"])
+        if getVarStrand(variant) == "-":
+            varExonStart = int(exonBounds[varExonNum]["exonStart"])
+            # newSplicePos -1 to account for RefSeq numbering which starts to the right of the first base
+            exonLength = varExonStart - (newSplicePos - 1)
+        else:
+            varExonEnd = int(exonBounds[varExonNum]["exonEnd"])
+            refExonLength = getRefExonLength(variant)
+            # need to compare to refExonLength because of + strand gene
+            # newSplicePos -1 to account for RefSeq numbering which starts to the right of the first base
+            exonLength = refExonLength - (varExonEnd - (newSplicePos - 1))
+        return exonLength
+
+def compareRefAltExonLengths(refLength, altLength):
+    '''
+    Compares ref and alt exon lengths
+    If both exon lengths % 3 are equal, then both ref and alt have the same reading frame
+    Returns true if both ref and alt exon have the same reading frame, false otherwise
+    '''
+    if refLength % 3 == altLength % 3:
+        return True
+    else:
+        return False
+
+def isSplicingWindowInFrame(variant):
+    '''
+    Given a variant, determines ref and alt exon length and compares them
+    If ref and alt exon are in the same reading frame, returns True
+    '''
+    refLength = getRefExonLength(variant)
+    altLength = getAltExonLength(variant)
+    inFrame = compareRefAltExonLengths(refLength, altLength)
+    if inFrame == True:
+        return True
+    return False
+
+def compareDeNovoWildTypeSplicePos(variant):
+    '''
+    Given a variant, compares de novo splicing position with wild-type splicign position
+    If distance between de novo and wild-type donors is divisible by 3, returns True
+    returns False otherwise
+    '''
+    if varInExon(variant) == True:
+        varStrand = getVarStrand(variant)
+        varExonNum = getVarExonNumberSNS(variant)
+        refExonBounds = getExonBoundaries(variant)
+        wildTypeSplicePos = refExonBounds[varExonNum]["exonEnd"]
+        slidingWindowInfo = getMaxMaxEntScanScoreSlidingWindowSNS(variant)
+        deNovoSplicePos = getNewSplicePosition(variant["Pos"], varStrand,
+                                               slidingWindowInfo["varWindowPosition"], slidingWindowInfo["inFirstThree"])
+        if varStrand == "+":
+            distanceBetween = wildTypeSplicePos - deNovoSplicePos
+        else:
+            # +1 because of RefSeq numbering
+            distanceBetween = deNovoSplicePos - (wildTypeSplicePos + 1)
+        if distanceBetween % 3 == 0:
+            return True
+        else:
+            return False
+        
+def determineSpliceRescueSNS(variant, boundaries):
```

Comment:
This function name seems off.  As well as determining splice rescue, this function assigns the prior and predicted qualitative class.  How about a broader name?